tasks/connectors/ms_defender_tvm/ms_defender_tvm.rb

```diff
@@ -133,9 +132,19 @@ def run(opts)
             machine_id = vuln.fetch("deviceId")
             fqdn = vuln.fetch("deviceName")
 
+            ip_json_response = get_ip(machine_id) unless machine_data.key?(machine_id)
+            unless ip_json_response.nil?
+              machine_data[machine_id] = ["ip_address" => ip_json_response["lastIpAddress"],
+                                          "tags" => ip_json_response["machineTags"],
+                                          "status" => ip_json_response["healthStatus"]]
+            end
             # Get the asset details & craft them into a hash
+
+            next unless machine_data[machine_id].first.dig("status") == "Active"
```

Comment:
I think, this would error with `undefined method `first' for nil:NilClass (NoMethodError)` if `ip_json_response` is `nil`